pkg/remoteclient/remoteclient.go

```diff
@@ -140,7 +140,7 @@ func InitialURL(c client.Client, cd *hivev1.ClusterDeployment) (string, error) {
 // the ClusterDeployment to determine if the remote cluster is reachable.
 func Unreachable(cd *hivev1.ClusterDeployment) (unreachable bool, lastCheck time.Time) {
 	cond := utils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
-	if cond.Status == corev1.ConditionUnknown {
+	if cond == nil || cond.Status == corev1.ConditionUnknown {
```

Comment:
This seems unrelated, should be a separate PR.

Reply:
We actually crash if you adopt without this, I will retitle the PR.